packages/simcore-sdk/src/simcore_sdk/node_ports_common/data_items_utils.py

```diff
@@ -20,12 +22,9 @@ def create_simcore_file_id(
     return SimcoreS3FileID(f"{clean_path}")
 
 
-_INTERNAL_DIR = Path(tempfile.gettempdir(), "simcorefiles")
-
-
 def create_folder_path(key: str) -> Path:
-    return Path(_INTERNAL_DIR, f"{threading.get_ident()}", key)
+    return _TMP_SIMCOREFILES / f"{uuid4()}" / key

```

Comment:
Just to be sure, how is this seeded? In case it's still running in multiple threads/processes.

Reply:
This seems to point to the fact things might not always be multiprocessing safe: https://docs.python.org/3/library/uuid.html#uuid.UUID.is_safe

Reply:
In short, nothing to worry about.

The key here is `uuid4()` which is generated by the same process. The likelihood of a collision is basically not existing in real life.
For a better explanation ask you favourite smarter chatbot about it `Likelihood of uuid4 collisions generated by the same process`.

Reply:
As long as it's in the same process I agree.

Reply:
@pcrespov I see no direct benefit in your change. I would also have to change the downstream functions.
Since the current solution fixes the issue, I think putting more effort here is not justified.

Reply:
ok. discussed offline. thx